subscription.go

```diff
@@ -2,13 +2,15 @@ package pubsub
 
 import (
 	"context"
+	protocol "github.com/libp2p/go-libp2p-protocol"
 )
 
 type Subscription struct {
-	topic    string
-	ch       chan *Message
-	cancelCh chan<- *Subscription
-	err      error
+	topic         string
+	topicProtocol protocol.ID
```

Comment:
I don't think this belongs here.

Reply:
I disagree, there is in PubSub a declaration `type SubOpt func(sub *Subscription) error` that is currently almost unusable because `Subscription` has no fields to work with.

I assume that pattern taken with both `Option` (for PubSub) and `SubOpt` (for Subscriptions) is designed to allow adding fields to the struct, and functionality as option wrappers, without breaking the struct contract/interface. Is that incorrect, if so how is `SubOpt` supposed to be used and where would you put the relevant topic/Subscription information?